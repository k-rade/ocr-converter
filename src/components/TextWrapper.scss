.text-wrapper {
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  width: 100%;
  position: relative;
  min-height: 11rem;

  &-heading {
    align-self: flex-start;
    margin-bottom: 0.5rem;
    color: var(--color-gray-900);
    font-weight: 500;
  }

  &-actions {
    display: flex;
    gap: 1rem;
    margin-top: 1rem;
  }

  &-output {
    background-color: var(--color-gray-200);
    border: 2px solid var(--color-gray-600);
    padding: 1.5rem;
    height: 100%;
    width: 100%;
    border-radius: 1px;
    box-sizing: border-box;
    resize: none;
    font-family: var(--font);
    font-size: 0.875rem;
    overflow-wrap: break-word;

    &:focus {
      outline: none;
    }
  }

}