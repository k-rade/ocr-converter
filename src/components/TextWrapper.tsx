import { copyText, downloadTextFile } from "../util/OCRConverterUtils";
import IconButton from "./IconButton";
import { RiFileCopyFill, RiFileDownloadFill } from "react-icons/ri";

interface Props {
  text?: string;
  initNewConversion: () => void;
}

const TextWrapper = ({ text = "", initNewConversion }: Props) => (
  <div className="top-0 left-0 bg-white w-full">
    <div className="flex items-end justify-between mb-4 text-lg font-thin text-gray-500">
      Output
      <button
        onClick={initNewConversion}
        type="button"
        className="py-2 px-4 bg-indigo-500 hover:bg-indigo-600 text-white text-sm font-medium rounded"
      >
        New conversion
      </button>
    </div>

    <textarea
      disabled
      value={text}
      className="w-full border-b-2 bg-gray-50 p-2 resize-none"
      placeholder="Output text"
      rows={11}
    />
    <div className="mt-1 flex justify-end gap-4">
      <IconButton
        label="Copy text"
        icon={
          <RiFileCopyFill
            data-testid="copy-icon"
            className="fill-gray-400 hover:fill-indigo-500"
            size="1.25rem"
          />
        }
        onClick={() => copyText(text)}
        {...(!text && { disabled: true })}
      />
      <IconButton
        label="Download"
        icon={
          <RiFileDownloadFill
            data-testid="download-icon"
            className="fill-gray-400 hover:fill-indigo-500"
            size="1.25rem"
          />
        }
        onClick={() => downloadTextFile(text)}
        {...(!text && { disabled: true })}
      />
    </div>
  </div>
);
export default TextWrapper;
