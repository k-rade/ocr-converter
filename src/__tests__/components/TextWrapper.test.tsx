import { fireEvent, render } from "@testing-library/react";
import TextWrapper from "../../components/TextWrapper";

Object.assign(navigator, {
  clipboard: {
    writeText: () => {},
  },
});

describe("TestWrapper", (): void => {
  it("should render", (): void => {
    const { baseElement } = render(<TextWrapper text="text" />);

    expect(baseElement).toMatchSnapshot();
  });

  it("should copy text to clipboard", (): void => {
    jest.spyOn(navigator.clipboard, "writeText");
    const { getByText } = render(<TextWrapper text="text" />);

    fireEvent.click(getByText("copy.svg"));

    expect(navigator?.clipboard?.writeText).toHaveBeenNthCalledWith(1, "text");
  });
});
