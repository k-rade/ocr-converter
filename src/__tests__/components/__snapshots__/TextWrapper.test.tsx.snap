// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestWrapper should render 1`] = `
<body>
  <div>
    <div
      class="top-0 left-0 bg-white w-full"
    >
      <div
        class="flex items-end justify-between mb-4 text-lg font-thin text-gray-500"
      >
        Output
        <button
          class="py-2 px-4 bg-indigo-500 hover:bg-indigo-600 text-white text-sm font-medium rounded"
          type="button"
        >
          New conversion
        </button>
      </div>
      <textarea
        class="w-full border-b-2 bg-gray-50 p-2 resize-none"
        disabled=""
        placeholder="Output text"
        rows="11"
      />
      <div
        class="mt-1 flex justify-end gap-4"
      >
        <div
          class="relative"
        >
          <button
            disabled=""
            type="button"
          >
            <svg
              class="fill-gray-400 hover:fill-indigo-500"
              data-testid="copy-icon"
              fill="currentColor"
              height="1.25rem"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1.25rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M7 6V3a1 1 0 0 1 1-1h12a1 1 0 0 1 1 1v14a1 1 0 0 1-1 1h-3v3c0 .552-.45 1-1.007 1H4.007A1.001 1.001 0 0 1 3 21l.003-14c0-.552.45-1 1.007-1H7zm2 0h8v10h2V4H9v2z"
                />
              </g>
            </svg>
          </button>
          <span
            class="absolute hidden bg-gray-800 text-white rounded p-1.5 px-3 text-xs -left-9 bottom-8 whitespace-nowrap"
          >
            Copy text
          </span>
        </div>
        <div
          class="relative"
        >
          <button
            disabled=""
            type="button"
          >
            <svg
              class="fill-gray-400 hover:fill-indigo-500"
              data-testid="download-icon"
              fill="currentColor"
              height="1.25rem"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1.25rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M16 2l5 5v14.008a.993.993 0 0 1-.993.992H3.993A1 1 0 0 1 3 21.008V2.992C3 2.444 3.445 2 3.993 2H16zm-3 10V8h-2v4H8l4 4 4-4h-3z"
                />
              </g>
            </svg>
          </button>
          <span
            class="absolute hidden bg-gray-800 text-white rounded p-1.5 px-3 text-xs -left-9 bottom-8 whitespace-nowrap"
          >
            Download
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;
